tizen-api/include/tizen-api.h

```diff
@@ -0,0 +1,285 @@
+
+/**
+ * @brief Stop the pipeline
+ * @detail The pipeline handle returned by nns_construct_pipeline (pipe) is stopped.
+ * @since_tizen 5.5
+ * @param[in] pipe The pipeline to be stopped.
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_stop (nns_pipeline_h *pipe);
+
+/****************************************************
+ ** NNStreamer Pipeline Sink/Src Control           **
+ ****************************************************/
+/**
+ * @brief Callback for sink (tensor_sink) of nnstreamer pipelines (nnstreamer's output)
+ * @detail If an application wants to accept data outputs of an nnstreamer stream, use this callback to get data from the stream. Note that the buffer may be deallocated after the return and this is synchronously called. Thus, if you need the data afterwards, copy the data to another buffer and return fast. Do not hold too much time in the callback. It is recommended to use very small tensors at sinks.
+ * @since_tizen 5.5
+ * @param[in] buf The contents of the tensor output (a single frame. tensor/tensors).
+ * @param[in] size The size of the buffer.
+ * @param[in] tensorinfo The cardinality, dimension, and type of given tensor/tensors.
+ * @param[in/out] pdata User Application's Private Data
+ */
+typedef void (*nns_sink_cb)
+(const char *buf, size_t size, const GstTensorsInfo *tensorinfo, void *pdata);
+
+/**
+ * @brief Register a callback for sink (tensor_sink) of nnstreamer pipelines.
+ * @since_tizen 5.5
+ * @param[in] pipe The pipeline to be attached with a sink node.
+ * @param[in] sinkname The name of sink node, described with nns_pipeline_construct().
+ * @param[in] cb The function to be called by the sink node.
+ * @return The nns_sink_handle. Returns NULL if fails.
+ */
+nns_sink_handle nns_pipeline_sink_register
+(nns_pipeline_h *pipe, const char *sinkname, nns_sink_cb cb);
+
+/**
+ * @brief Unregister a callback for sink (tensor_sink) of nnstreamer pipelines.
+ * @since_tizen 5.5
+ * @param[in] The nns_sink_handle to be unregistered (destroyed)
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_sink_unregister (nns_sink_handle h)
+
+/**
+ * @brief Get a handle to operate as a src node of nnstreamer pipelines.
+ * @since_tizen 5.5
+ * @param[in] pipe The pipeline to be attached with a src node.
+ * @param[in] srcname The name of src node, described with nns_pipeline_construct().
+ * @param[out] tensorinfo The cardinality, dimension, and type of given tensor/tensors.
+ * @return The nns_src_handle. Returns NULL if fails.
+ */
+nns_src_handle nns_pipeline_src_gethandle
+(nns_pipeline_h *pipe, const char *srcname, GstTensorsInfo *tensorinfo);
+
+/**
+ * @brief Register a callback
+ * @param[out] bufpolicy The policy of buf deallocation.
+ * @return 0 on success (buf is filled). otherwise a negative error value.
+ * @retval #NNS_ERROR_NONE Successful
+ */
```

Comment:
Above comments for doxygen look slightly different from the function prototype.

Reply:
Updated